src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -504,6 +504,24 @@ internal static class Assembly
                 .CreateDelegate<Func<System.Reflection.Assembly, string, bool, bool, Type>>();
         }
 
+        internal static class AssemblyName
+        {
+            private const string TypeName = "System.Reflection.AssemblyName";
+
+            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            private static readonly PropertyInfo CodeBase = Type
+                .GetTypeInfo()
+                .GetDeclaredProperty(nameof(CodeBase));
```

Comment:
In that case, a better alternative might be to require the workspace host to provide the implementation of the host analyzer assembly loader. [GetHostDiagnosticAnalyzerPackages](http://source.roslyn.io/Microsoft.CodeAnalysis.Features/R/c64619a460b7e1ad.html) API implemented in VS Workspace host should provide the assembly loader for loading the host analyzers (VSIX analyzers). We can then use the prior non-portable implementation of the assembly loader in the VS layer. In future, if we have a CoreCLR based workspace host, that host would be responsible for providing an appropriate analyzer loader. @srivatsn @heejaechang what do you think?